### ISSUE TITLE
Wrong column reported in block-no-empty

### ISSUE THREAD

**Issue by Arcanemagus** [stylelint/stylelint#1889]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where `block-no-empty` reports the wrong column for certain SugarSS files.

> Which rule, if any, is this issue related to?

`block-no-empty`

> What CSS is needed to reproduce this issue?

```sugarss
.profile-identities_item
  min-height: 25px
  .profile-identities_item.is-facebook
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "block-no-empty": true
  }
}
```

Note: A collection of files that reproduces this is available here: [linter-stylelint_GH260.zip](https://github.com/AtomLinter/linter-stylelint/files/457552/linter-stylelint_GH260.zip)

> Which version of stylelint are you using?

`stylelint@7.2.0` and `linter-stylelint@3.2.1`.

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint foo.sss --syntax sugarss`, and through the Node API (Atom `linter-stylelint` package).

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SugarSS.

> What did you expect to happen?

The proper column being reported.

> What actually happened (e.g. what warnings or errors you are getting)?

A column beyond the end of the line is being reported. Stylelint reports an issue on column 41 (1-indexed) of a line that only has 39 columns.

![](https://cloud.githubusercontent.com/assets/427137/18286909/b9313654-7429-11e6-8a28-ef17e5b37dcb.png)

I'm _guessing_ that this is a mapping issue between the SugarSS, the CSS that `stylelint` actually runs on, and then back to the original SugarSS, but I am not positive.

This was originally filed by @ai as https://github.com/AtomLinter/linter-stylelint/issues/260.

**Comment by Arcanemagus** [comment]
Hmmm, looks like I _already_ reported this once before 😛.

It was reported in https://github.com/stylelint/stylelint/issues/1544, and suppoedly fixed in https://github.com/stylelint/stylelint/pull/1547, but it looks like the issue is back.

**Comment by evilebottnawi** [comment]
@Arcanemagus accepted, thanks for reporting

**Comment by Arcanemagus** [comment]
Looks like another case is the following:

```sugarss
body
  padding: 5% 2% 5%

.
```

It reports an error on `4:4`, when `4:1` is the last column of the file.

**Comment by davidtheclark** [comment]
Fixed by #1978.
